core/src/main/java/org/apache/iceberg/io/SortedPosDeleteWriter.java

```diff
@@ -167,7 +167,9 @@ private void flushDeletes() {
         List<PosRow<T>> positions = posDeletes.get(wrapper.set(path));
         positions.sort(Comparator.comparingLong(PosRow::pos));
 
-        positions.forEach(posRow -> closeableWriter.delete(path, posRow.pos(), posRow.row()));
+        PositionDelete<T> posDelete = PositionDelete.create();
+        positions.forEach(
+            posRow -> closeableWriter.write(posDelete.set(path, posRow.pos(), posRow.row())));
```

Comment:
@nastra, this changes the behavior to create a new `PositionDelete` for every delete rather than reusing the same one. Can you follow up with a fix to restore the old behavior?

Reply:
I'll make sure to be more careful when reviewing such behavioral changes and I opened https://github.com/apache/iceberg/pull/5896 to address that